src/apply_conf.rs

```diff
@@ -181,31 +181,35 @@ fn copy_connection_files(
         .to_str()
         .ok_or_else(|| anyhow!("Determining host config path"))?;
 
-    for interface in &host.interfaces {
+    for interface in host.interfaces {
         info!("Processing interface '{}'...", &interface.logical_name);
+        let connections = &interface.connection_ids.unwrap_or_else(Vec::new);
```

Comment:
We should first take a look at whether an option is necessary but this current business logic essentially means that we'd iterate over nothing (empty vector) if there are `interface.connection_ids` is None.